pkg/image/fetcher.go

```diff
@@ -171,8 +178,28 @@ func (f *Fetcher) fetchDaemonImage(name string) (imgutil.Image, error) {
 	return image, nil
 }
 
-func (f *Fetcher) fetchRemoteImage(name string) (imgutil.Image, error) {
-	image, err := remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+func (f *Fetcher) fetchRemoteImage(name string, target *dist.Target) (imgutil.Image, error) {
+	var (
+		image imgutil.Image
+		err   error
+	)
+
+	if target == nil {
+		image, err = remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+	} else {
+		platform := imgutil.Platform{OS: target.OS, Architecture: target.Arch}
+		// when targeting a registry, we need to use variant and osVersion to hit the correct image
```

Comment:
You mean the following:

- In the registry, we have `linux/amd64` and `linux/arm64` for example
- In target we have `linux/arm64/v5/ubuntu`

If we configure  `linux/arm64/v5/ubuntu`, is it going to hit `linux/arm64` ?

Reply:
Yeah

Reply:
I did the following test:

## Case 1

- I created some local builders, based on `alpine` and image index looks like this:
```json
"manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:22be7352ae0e2f030507526e5547d16673b7bc92c7f84cd75d8ed9856fb4018f",
      "size": 861,
      "platform": {
        "architecture": "arm",
        "os": "linux",
        "variant": "v7"
     }
]
```

I tried to create a multi-platform builder, but in my `builder.toml`, it matches exactly my `build` and `run` image in the registry.

```toml
[[targets]]
os = "linux"
arch = "arm"
variant = "v7"
```

In this case, everything is fine. the builder is created.

## Case 2

I added a distribution in my `builder.toml` 

```toml
[[targets]]
os = "linux"
arch = "arm"
variant = "v7"
[[targets.distros]]
name = "alpine"
version = "3.19.3"
```
In this case, the builder was also created, and when I check the config file the layers `DiffIds` matches with the `linux/arm/v7` manifest. It looks like `osVersion` is not used to match the registry.  

It looks like we can remove it, and your concern with creating duplicated images for each distribution could be solved, maybe we can do that after releasing this? if it is required?

Reply:
I mean, I think it makes sense (if there is a more specific image) to pick `linux/arm64/v5/ubuntu` if it exists. BUT we should fall back to the less specific version because it's likely not all (many?) images have all those fields set (kind of like where we check for the error `image with reference <image> was found but does not match the specified platform: wanted linux/amd64, actual: linux`).

Reply:
Well, just how it is right now, it doesn't fail, and it fall back to `linux/arm/v7` so I think we are fine on how it is implemented.